cloud/aws/services/cloudformation/bootstrap.go

```diff
@@ -0,0 +1,340 @@
+					"ec2:CreateTags",
+					"ec2:CreateVolume",
+					"ec2:ModifyInstanceAttribute",
+					"ec2:ModifyVolume",
+					"ec2:AttachVolume",
+					"ec2:AuthorizeSecurityGroupIngress",
+					"ec2:CreateRoute",
+					"ec2:DeleteRoute",
+					"ec2:DeleteSecurityGroup",
+					"ec2:DeleteVolume",
+					"ec2:DetachVolume",
+					"ec2:RevokeSecurityGroupIngress",
+					"ec2:DescribeVpcs",
+					"elasticloadbalancing:AddTags",
+					"elasticloadbalancing:AttachLoadBalancerToSubnets",
+					"elasticloadbalancing:ApplySecurityGroupsToLoadBalancer",
+					"elasticloadbalancing:CreateLoadBalancer",
+					"elasticloadbalancing:CreateLoadBalancerPolicy",
+					"elasticloadbalancing:CreateLoadBalancerListeners",
+					"elasticloadbalancing:ConfigureHealthCheck",
+					"elasticloadbalancing:DeleteLoadBalancer",
+					"elasticloadbalancing:DeleteLoadBalancerListeners",
+					"elasticloadbalancing:DescribeLoadBalancers",
+					"elasticloadbalancing:DescribeLoadBalancerAttributes",
+					"elasticloadbalancing:DetachLoadBalancerFromSubnets",
+					"elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
+					"elasticloadbalancing:ModifyLoadBalancerAttributes",
+					"elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+					"elasticloadbalancing:SetLoadBalancerPoliciesForBackendServer",
+					"elasticloadbalancing:AddTags",
+					"elasticloadbalancing:CreateListener",
+					"elasticloadbalancing:CreateTargetGroup",
+					"elasticloadbalancing:DeleteListener",
+					"elasticloadbalancing:DeleteTargetGroup",
+					"elasticloadbalancing:DescribeListeners",
+					"elasticloadbalancing:DescribeLoadBalancerPolicies",
+					"elasticloadbalancing:DescribeTargetGroups",
+					"elasticloadbalancing:DescribeTargetHealth",
+					"elasticloadbalancing:ModifyListener",
+					"elasticloadbalancing:ModifyTargetGroup",
+					"elasticloadbalancing:RegisterTargets",
+					"elasticloadbalancing:SetLoadBalancerPoliciesOfListener",
+					"iam:CreateServiceLinkedRole",
+					"kms:DescribeKey",
+				},
+			},
+		},
+	}
+}
+
+// From https://github.com/kubernetes/cloud-provider-aws
+func cloudProviderNodeAwsPolicy() *iam.PolicyDocument {
+	return &iam.PolicyDocument{
+		Version: iam.CurrentVersion,
+		Statement: []iam.StatementEntry{
+			{
+				Effect:   iam.EffectAllow,
+				Resource: iam.Resources{"*"},
+				Action: iam.Actions{
+					"ec2:DescribeInstances",
+					"ec2:DescribeRegions",
+					"ecr:GetAuthorizationToken",
+					"ecr:BatchCheckLayerAvailability",
+					"ecr:GetDownloadUrlForLayer",
+					"ecr:GetRepositoryPolicy",
+					"ecr:DescribeRepositories",
+					"ecr:ListImages",
+					"ecr:BatchGetImage",
+				},
+			},
+		},
+	}
+}
+
+// CreateBootstrapStack creates a AWS CloudFormation stack to initialise
+// IAM policy
+func CreateBootstrapStack() error {
+	stackName := "cluster-api-provider-aws-sigs-k8s-io"
+	sess := session.New()
+	svc := cfn.New(sess)
+
+	template := BootstrapTemplate()
+	yaml, err := template.YAML()
+	if err != nil {
+		return err
```

Comment:
please wrap errors with `errors.Wrap(err, "failed to generate cloudformation YAML")` or similar message. This makes it much easier to debug both during development and a user submits a ticket.